lib/view/crypto_view.dart

```diff
@@ -0,0 +1,301 @@
+import 'package:autonomy_flutter/screen/bloc/scan_wallet/scan_wallet_bloc.dart';
+import 'package:autonomy_flutter/screen/bloc/scan_wallet/scan_wallet_state.dart';
+import 'package:autonomy_flutter/util/constants.dart';
+import 'package:autonomy_flutter/util/string_ext.dart';
+import 'package:autonomy_flutter/util/style.dart';
+import 'package:autonomy_flutter/util/wallet_utils.dart';
+import 'package:autonomy_flutter/view/primary_button.dart';
+import 'package:autonomy_flutter/view/radio_check_box.dart';
+import 'package:autonomy_flutter/view/responsive.dart';
+import 'package:autonomy_theme/autonomy_theme.dart';
+import 'package:collection/collection.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:libauk_dart/libauk_dart.dart';
+
+class AddAddressToWallet extends StatefulWidget {
+  final List<AddressInfo> addresses;
+  final List<String> importedAddress;
+  final bool scanNext;
+  final Function(List<AddressInfo> addesses)? onImport;
+  final WalletType walletType;
+  final WalletStorage wallet;
+  final Function? onSkip;
+
+  const AddAddressToWallet(
+      {Key? key,
+      required this.addresses,
+      required this.importedAddress,
+      this.scanNext = true,
+      this.onImport,
+      required this.walletType,
+      required this.wallet,
+      this.onSkip})
+      : super(key: key);
+
+  @override
+  State<AddAddressToWallet> createState() => _AddAddressToWalletState();
+}
+
+class _AddAddressToWalletState extends State<AddAddressToWallet> {
+  List<AddressInfo> addresses = [];
+  List<AddressInfo> selectedAddresses = [];
+  int index = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    addresses = widget.addresses;
+    if (addresses.isEmpty) _callBloc(false);
+  }
+
+  @override
+  void dispose() {
+    addresses = [];
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return BlocConsumer<ScanWalletBloc, ScanWalletState>(
+        listener: (context, scanState) {
+      if (scanState.addresses.isNotEmpty && !scanState.isScanning) {
+        setState(() {
+          addresses = scanState.addresses;
```

Comment:
why need to reassign to a local variable here. Can we use the `state.addresses` directly

Reply:
there is case that use addresses from parameter